### ISSUE TITLE
Text not centred in Android M 

### ISSUE THREAD

**Issue by mrbrentkelly** [BCGDV-ASIA/android-widget-ticktock#1]
When using TickTockView in Android M the Text is not centred correctly...

**Android 4.4.2 KitKat**
<img width="208" alt="screen shot 2016-05-29 at 10 41 36" src="https://cloud.githubusercontent.com/assets/5814579/15632461/f40268e6-2589-11e6-9869-cdc85271d580.png">

**Android 6.0 Marshmallow**
<img width="237" alt="screen shot 2016-05-29 at 10 40 29" src="https://cloud.githubusercontent.com/assets/5814579/15632460/f3fff656-2589-11e6-8dd4-ffa48becb252.png">

**Comment by alphamu** [comment]
Cab you please post your layout XML? We have test out on M and it looks fine. Are there any styles that could be throwing it off? TickTickView extends TextView so any default styles on TextView could be affecting it. Though, just looks like auto sizing is not working.

**Comment by mrbrentkelly** [comment]
Hey here's the Layout XML...

``` xml
<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="@color/colorAccent">
    <com.bcgdv.asia.lib.ticktock.TickTockView
        android:id="@+id/view_ticktock_countdown"
        android:layout_width="200dp"
        android:layout_height="200dp"
        android:layout_marginTop="16dp"
        android:layout_gravity="center_horizontal"
        app:tickAutoFitText="true"
        app:tickMiddleColor="#333333"
        app:tickMoveCounterClockwise="true"
        app:tickTextColor="#CCCCCC"
        app:tickDotRadius="6dp"
        app:tickEmptyRingColor="@android:color/white"
        app:tickFillRingColor="@color/colorPrimary"
        app:tickRingThickness="3dp" />
</FrameLayout>
```

Just to be sure its not related to any of my styling I've used the sample XML and code sample in your README in a brand new Activity with no other styling. Running on Nexus 5X API 23 (6.0) emulator in Android Studio.

<img width="268" alt="screen shot 2016-05-29 at 22 53 17" src="https://cloud.githubusercontent.com/assets/5814579/15636238/2bd5b578-25f0-11e6-87ad-9f3a03f58538.png">

**Comment by alphamu** [comment]
Hey Brent, thanks for you patience.

I've tried the same app, on the Android Studio Emulator 2.0 (Running M and N) and on A Nexus 5X device and haven't found an issue. Look at the screens below.

There seems to be something in your use case that is breaking things. There is a known issue of reserving space with the centered text, i.e. if I start with "1:1:1" this takes up a lot less space than "1:1:55" so when "1:1:11" become  "1:1:55" the text looks broken. The best way to fix this is to always use 2 digits to represent time "00:00:00" or at least set this initially to reserve space. I'll see if this problem can be solved, though it seems hard since I don't really know what the use case for the centre text could be. That being said, doesn't look like the problem you're seeing. 

I've tried referencing the library directly as well as including it as a jcenter dependency, I don't see any issues. If you can email me your project/app, that might help.

## Nexus 5X phone running 6.0.1

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15637025/f3543860-2652-11e6-94c4-1ecdaaad9b31.png" />

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15636764/c1521ea0-264c-11e6-98f9-fb004d864729.png" />

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15637030/1c293a24-2653-11e6-9fc8-ab7b10ca26f3.png"/>

## Android M emulator (Nexus 5X)

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15636769/d7cbf6a6-264c-11e6-96a6-4a36844c205b.png" />

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15636770/d7cdbcfc-264c-11e6-8084-878bf5849f9b.png"/>

##Android N Emulator (Nexus 5X)

<img  width="50%" height="50%" src="https://cloud.githubusercontent.com/assets/844940/15636782/f47a9fdc-264c-11e6-80dc-7a47b9d3c8cc.png"/>

**Comment by mrbrentkelly** [comment]
Hey thanks for looking into this. I've done a bit more digging this morning and it seems like the issue occurs when I use 'Instant Run' or [JRebel](https://zeroturnaround.com/software/jrebel-for-android/) to apply code changes. So it seems like the widget can't handle hotswaps?

To reproduce... you run the app on a clean install the text will appear fine, make a small code change, then run the app again using Instant Run (make sure you get the lightning symbol <img width="35" alt="screen shot 2016-05-30 at 09 44 59" src="https://cloud.githubusercontent.com/assets/5814579/15644493/4d2d9ae6-264b-11e6-8d31-e91b6467ca5d.png">) and the text will be off center.

**Comment by alphamu** [comment]
@mrbrentkelly sorry it took so long, but the nature of the bug made it a low priority.

This issue should be resolved commit bd105e5816ab4e20d7ac54d6f0c7b582ac2fc93a and is available in release 1.2.

**Comment by danieltigse** [comment]
I have the same error running Android 6.0.2 Samsung Galaxy S6 and with the 1.2 version. And issue occurs even without instant run.

**Comment by alphamu** [comment]
Is that again with Hotswap?

**Comment by danieltigse** [comment]
It occurs without hotswap. I also put a breakpoint in the fitText method and never entered there.

**Comment by danieltigse** [comment]
Another clue: onSizeChanged method is never called.
EDIT:
onSizeChanged is called BUT mText is always null

**Comment by danieltigse** [comment]
I think it is a synchronization problem: I solve it by adding a postdelay after called method start od the TickView

``` java
final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                if (countDownHours != null) {
                    countDownHours.start(start, end);
                }
            }
        }, 100);
```

**Comment by alphamu** [comment]
More likely that the view hierarchy has not finished loading  when you try to start the widget. This can happen for complex layouts, you may see something similar when you try to do shared element transitions. You should probably register a view hierarchy listener and start the widget after the layout completes.

**Comment by danieltigse** [comment]
Ok I'll try it thanks.

**Comment by raylenmoore** [comment]
one reason for the issue is when start TickTockView the mText is null:
```
        //Fix for issue #1 where hot deploying would result in text being rendered incorrectly.
        if (!TextUtils.isEmpty(mText) && (mTextBounds.width() == 0 || mTextBounds.height() == 0) && mTextSize > 0) {
            //When code is hot deployed,
            fitText(mText);
        }
```
add the following line in the layout to offer the initialized text:
app:tickText="00:00"
